server/coordinator/procedure/scatter.go

```diff
@@ -0,0 +1,216 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+package procedure
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/CeresDB/ceresdbproto/pkg/clusterpb"
+	"github.com/CeresDB/ceresmeta/pkg/log"
+	"github.com/CeresDB/ceresmeta/server/cluster"
+	"github.com/CeresDB/ceresmeta/server/coordinator/eventdispatch"
+	"github.com/CeresDB/ceresmeta/server/id"
+	"github.com/looplab/fsm"
+	"github.com/pkg/errors"
+)
+
+const (
+	eventScatterPrepare = "EventScatterPrepare"
+	eventScatterFailed  = "EventScatterFailed"
+	eventScatterSuccess = "EventScatterSuccess"
+
+	stateScatterBegin   = "StateScatterBegin"
+	stateScatterWaiting = "StateScatterWaiting"
+	stateScatterFinish  = "StateScatterFinish"
+	stateScatterFailed  = "StateScatterFailed"
+
+	defaultTimeInterval = time.Second * 3
+)
+
+var (
+	scatterEvents = fsm.Events{
+		{Name: eventScatterPrepare, Src: []string{stateScatterBegin}, Dst: stateScatterWaiting},
+		{Name: eventScatterSuccess, Src: []string{stateScatterWaiting}, Dst: stateScatterFinish},
+		{Name: eventScatterFailed, Src: []string{stateScatterWaiting}, Dst: stateScatterFailed},
+	}
+	scatterCallbacks = fsm.Callbacks{
+		eventScatterPrepare: scatterPrepareCallback,
+		eventScatterFailed:  scatterFailedCallback,
+		eventScatterSuccess: scatterSuccessCallback,
+	}
+)
+
+func scatterPrepareCallback(event *fsm.Event) {
+	request := event.Args[0].(*ScatterCallbackRequest)
+	c := request.cluster
+	ctx := request.ctx
+
+	for {
+		time.Sleep(defaultTimeInterval)
+		if uint32(c.GetNodesSize()) < c.GetClusterMinNodeCount() {
+			log.Warn("wait for cluster node register")
```

Comment:
It will be better to define a method called `waitForNodesReady` to do such things.